pkg/sql/colexec/sort_tmpl.go

```diff
@@ -131,14 +133,23 @@ func newSingleSorter(
 // {{range .WidthOverloads}}
 
 type sort_TYPE_DIR_HANDLES_NULLSOp struct {
-	sortCol       _GOTYPESLICE
+	sortCol _GOTYPESLICE
+	// {{if .CanAbbreviate}}
+	abbreviatedSortCol []uint64
+	// {{end}}
 	nulls         *coldata.Nulls
 	order         []int
 	cancelChecker colexecutils.CancelChecker
 }
 
-func (s *sort_TYPE_DIR_HANDLES_NULLSOp) init(ctx context.Context, col coldata.Vec, order []int) {
+func (s *sort_TYPE_DIR_HANDLES_NULLSOp) init(
+	ctx context.Context, allocator *colmem.Allocator, col coldata.Vec, order []int,
+) {
 	s.sortCol = col.TemplateType()
+	// {{if .CanAbbreviate}}
+	allocator.AdjustMemoryUsage(memsize.Uint64 * int64(s.sortCol.Len()))
```

Comment:
I didn't add any code to "undo" this usage increase, because I'm assuming that when the sort operation cleans up it will automatically shrink all the added usage for this allocator, including this usage. Is that correct?